lib/devise/models/database_authenticatable.rb

```diff
@@ -37,6 +37,7 @@ def self.required_fields(klass)
 
       # Generates password encryption based on the given value.
       def password=(new_password)
+        self.encrypted_password_will_change! #trigger validations of the newly set password
```

Comment:
should we push this inside the `if @password.present?` from the `encrypted_password` assignment?